examples_utils/benchmarks/command_utils.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2022 Graphcore Ltd. All rights reserved.
+import re
+import logging
+
+
+def create_variants(benchmark_dict: dict) -> list:
+    """
+    Create all variants from a benchmark entry.
+
+    Notes
+    -----
+    There are two different ways to create variants from a benchmark entry:
+    Declarative and Matrix-style
+
+    With the first, when parameter values are listed, values for each parameter
+    are matched by their index in the list only, and variants are created for
+    each such combination. For example:
+        - batch_size: [3, 4, 5]
+        - gradient_accumulation: [150, 120, 90]
+    Will result in 3 variants as such:
+        - variant 1: batch_size=3, gradient_accumulation=150
+        - variant 2: batch_size=4, gradient_accumulation=120
+        - variant 3: batch_size=5, gradient_accumulation=90
+    The length of the shortest list determines the number of variants.
+
+    As for the second, when parameter values are provided in a dict with {},
+    variations are instead created by every possible combination of parameters.
+    For example:
+        - batch_size: {3, 4}
+        - gradient_accumulation: {150, 120}
+    Will result in 4 variants:
+        - variant 1: batch_size=3, gradient_accumulation=150
+        - variant 2: batch_size=3, gradient_accumulation=120
+        - variant 3: batch_size=4, gradient_accumulation=150
+        - variant 4: batch_size=4, gradient_accumulation=120
+    Dicts can have varying numbers of values, the above rule will still apply.
+
+    Parameters
+    ----------
+    benchmark_dict:
+        benchmark entry itself in yaml file
+
+    Returns
+    -------
+    variants: list
+        List of all variants from the given benchmark entry
```

Comment:
Can you please use google style docstrings

Reply:
Sure, can do.

Is this because we as a team have just chosen google style as a standard? Personally I prefer numpy style, but will stick to what the team has chosen,

Reply:
TBH it wasn't an app team decision. I choose google as that's what we use in popxl and popxl apps as standard and I want this repo to maintain some consistency